pyramid/config/__init__.py

```diff
@@ -903,6 +921,13 @@ def end(self):
         """
         return self.manager.pop()
 
+    def __enter__(self):
+        self.begin()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.end()
```

Comment:
I've thought a bit more about this and I think adding a `config.commit()` is the right answer here but it would probably be nice if we can avoid doing a commit if it's embedded in another context manager. @tseaver Good idea? Bad idea?